src/RageFileDriverDirectHelpers.cpp

```diff
@@ -28,36 +28,14 @@ RString DoPathReplace(const RString &sPath)
 #if defined(_WIN32)
 static bool WinMoveFileInternal( const RString &sOldPath, const RString &sNewPath )
 {
-	static bool Win9x = false;
-
-	/* Windows botches rename: it returns error if the file exists. In NT,
-	 * we can use MoveFileEx( new, old, MOVEFILE_REPLACE_EXISTING ) (though I
-	 * don't know if it has similar atomicity guarantees to rename). In
-	 * 9x, we're screwed, so just delete any existing file (we aren't going
-	 * to be robust on 9x anyway). */
-	if( !Win9x )
+	if( MoveFileEx(sOldPath, sNewPath, MOVEFILE_REPLACE_EXISTING) )
+		return true;
+	if( GetLastError() == ERROR_ALREADY_EXISTS )
```

Comment:
I like the previous way for the if statements without the nesting. It's easier to follow.